shimming-toolbox/shimmingtoolbox/files/NiftiMask.py

```diff
@@ -60,14 +60,12 @@ def load_mask(self, nif_target: NiftiTarget):
         else:
             raise ValueError("Mask must be in 3d or 4d")
 
-        if not np.all(nii_mask_target.shape == nif_target.shape) or not np.all(
-                nii_mask_target.affine == nif_target.affine):
-            logger.debug("Resampling mask on the target target")
-            nii_mask_target_soft = resample_from_to(nii_mask_target, nif_target.nii, order=1, mode='grid-constant')
-            tmp_mask = nii_mask_target_soft.get_fdata()
-            # Change soft mask into binary mask
-            tmp_mask = threshold(tmp_mask, thr=0.001, scaled_thr=True)
-            nii_mask_target = nib.Nifti1Image(tmp_mask, nii_mask_target_soft.affine, header=nii_mask_target_soft.header)
+        # Check if the mask needs to be resampled
+        if not np.all(nii_mask_target.shape == nif_target.shape) or not np.all(nii_mask_target.affine == nif_target.affine):
+            # Resample the mask on the target
+            logger.debug("Resampling mask on the target")
+            nii_mask_target = resample_from_to(nii_mask_target, nif_target, order=1, mode='grid-constant')
```

Comment:
It seems some errors slipped in when merging with master. Resample_from_to uses Nibabel objects not Nifs, but it got modified to nif in this PR.
```suggestion
            nii_mask_target = resample_from_to(nii_mask_target, nif_target.nii, order=1, mode='grid-constant')
```

Reply:
It's the only place in the merge conflict that I adapted the code manually. I checked all other modified lines and did not find any other problems. Your suggestion was implemented in commit [4077ce6](https://github.com/shimming-toolbox/shimming-toolbox/pull/581/commits/4077ce689ce769bdfa15543d79e600ebcfbc4821)